src/main/java/cascading/tap/hive/HivePartitionTap.java

```diff
@@ -123,6 +160,91 @@ public void sourceConfInit( FlowProcess<? extends Configuration> flowProcess, Co
     super.sourceConfInit( flowProcess, conf );
     }
 
+
+  @Override
+  public String[] getChildPartitionIdentifiers( FlowProcess<? extends Configuration> flowProcess, boolean fullyQualified ) throws IOException
+    {
+    Configuration conf = flowProcess.getConfig();
+    IMetaStoreClient metaStoreClient = null;
+    try
+      {
+      metaStoreClient = ( (HiveTap) getParent() ).getMetaStoreClientFactory().newInstance( conf );
+      HiveTableDescriptor descriptor = ( (HiveTap) getParent() ).getTableDescriptor();
+
+      List<Partition> partitions;
+      LOG.info("Getting partitions from the Hive metastore");
+      if ( selectedPartitionValues == null || selectedPartitionValues.isEmpty() )
+        {
+        partitions = metaStoreClient.listPartitions( descriptor.getDatabaseName(), descriptor.getTableName(), NO_LIMIT );
+        }
+      else
+        {
+        partitions = getSelectedPartitions( metaStoreClient, descriptor, conf );
+        }
+
+      return getPartitionPaths( partitions );
+      }
+    catch ( TException exception )
+      {
+      throw new CascadingException( exception );
+      }
+    finally
+      {
+      if ( metaStoreClient != null )
+        {
+        metaStoreClient.close();
+        }
```

Comment:
If MetaStoreClient is AutoClosable, we should start using try-with-ressources, since we left Java 6 behind.